iodata/test/test_basis.py

```diff
@@ -105,20 +105,39 @@ def test_shell_info_propertes():
     assert obasis.nbasis == 1 + 4 + 4 + 5 + 6 + 7 + 9
 
 
+def test_shell_validators():
+    Shell(0, [0, 0], ['c', 'c'], np.zeros(6), np.zeros((6, 2)))
```

Comment:
I put it there because I expect it not to cause any error unlike the small variations on the following lines. It should add a comment to clarify this. A line could also be added to test explicitly that it passes the validation:

```python
# The following line is constructs a Shell instance with valid arguments.
# It should not raise a TypeError.
shell = Shell(0, [0, 0], ['c', 'c'], np.zeros(6), np.zeros((6, 2)))
# Rerun the validators as a double check.
attr.validate(shell)
```

See https://www.attrs.org/en/stable/api.html#attr.validate This would be double checking in the sense that the validators get already called in the constructor, but double checking doesn't hurt.